src/main/java/bookapp/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package bookapp.service.impl;
+
+import bookapp.dto.user.UserRegistrationRequestDto;
+import bookapp.dto.user.UserResponseDto;
+import bookapp.exception.RegistrationException;
+import bookapp.mapper.UserMapper;
+import bookapp.model.Role;
+import bookapp.model.User;
+import bookapp.repository.role.RoleRepository;
+import bookapp.repository.user.UserRepository;
+import bookapp.service.UserService;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
+    private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("Can`t register user");
```

Comment:
Let’s  give a more informative message, also let’s  handle it in GlobalExceptionHandler and return 400 status for a user